pkg/controller/perconaservermongodb/finalizers.go

```diff
@@ -2,35 +2,58 @@ package perconaservermongodb
 
 import (
 	"context"
+
 	api "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1"
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	finalizerDeletePVC              = "delete-psmdb-pvc"
+	finalizerDeletePSMDBPodsInOrder = "delete-psmdb-pods-in-order"
+)
+
+var errWaitingTermination = errors.New("waiting pods to be deleted")
+
 func (r *ReconcilePerconaServerMongoDB) checkFinalizers(ctx context.Context, cr *api.PerconaServerMongoDB) (shouldReconcile bool, err error) {
-	shouldReconcile = true
-	finalizers := []string{}
+	shouldReconcile = false
```

Comment:
What do you think about just adding a new method to `api.PerconaServerMongoDB`, like `cr.getOrderedFinalizers()` and in there enforce the order we need? I feel like this would be a bit simpler and cleaner. Then here the only change would be to call `for _, f := range cr.GetOrderedFinalizers()` instead of `for _, f := range cr.GetFinalizers()`

This psmdb finelizers logic is a bit cumbersome and if we at least leave it as it is a win in my book :)

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1095/commits/77726a3cd9ce430472e7ff6c2049282d00143ef7